### ISSUE TITLE
NuPIC Needs Integration Test Illustrating Correct Output To Use

### ISSUE THREAD

**Issue by cogmission** [numenta/nupic#2684]
There needs to be an example developers can refer to which illustrates how interaction with the algorithms is accomplished. 

This issue author is proof of that need.

Being the most official porter of NuPIC code, and therefore the most object test for correct interpretation of NuPIC, the fact that I had no reference and mistakenly used the incorrect output when utilizing and implementing the interaction between algorithms - is proof of this need.

* The test should show getting data from Encoders, SpatialPoolers, TemporalMemories, Classifiers, Anomaly(s), AnomalyLikelihood(s) and feeding that data into the appropriate subsequent algorithm or algorithm(s) in the case where output can be sent into multiple subsequent algorithm types.

**Comment by breznak** [comment]
:+1: I've burned my fingers when getting some TP output and passing it manually to Anomaly! 

* this could be an example as well, instead of a i-test.
* I'd like to point out to https://github.com/numenta/nupic/issues/2536 which got kindof stalled, this would be one of the usecases.

**Comment by cogmission** [comment]
@breznak I agree, however I think that mentioning that here may be distracting. It's probably better to mention this Issue from #2536 rather than have a discussion of that here. This issue should remain pure so that it doesn't get coopted by off-topic conversation (as I have been guilty of doing too!)
